dags/app/finance_bot/udf.py

```diff
@@ -0,0 +1,97 @@
+import pygsheets
+import numpy as np
+from google.cloud import bigquery
+import pandas as pd
+import requests
+import os
+from app import discord
+
+
+session = requests.session()
+
+
+def main() -> None:
+    # read xls from google doc to df.
+    df_xls = read_google_xls_to_df()
+    # read bigquery to df.
+    df_bigquery = read_bigquery_to_df()
+    # check difference between 2 df
+    df_diff = df_difference(df_xls, df_bigquery)
+    # link to bigquery and write xls file
+    write_to_bigquery(df_diff)
+    # push to discord
+    webhook_url = os.getenv("discord_data_stratagy_webhook")
+    username = "財務機器人"
+    msg = refine_diff_df_to_string(df_diff)
+    discord.send_webhook_message(webhook_url, username, msg)
```

Comment:
just my two cents, but maybe we don't need to send any msg if there's no updates. Otherwise, that channel might be very noisy

Reply:
I have modify if msg is no data it didn't send msg